src/i2s/std.rs

```diff
@@ -800,279 +743,85 @@ impl<'d, Dir: I2sRxSupported + I2sTxSupported> I2sStdDriver<'d, Dir> {
     #[allow(clippy::too_many_arguments)]
-    pub fn new_rx<I2S: I2s>(
-        _i2s: impl Peripheral<P = I2S> + 'd,
-        config: config::StdConfig,
+    pub fn new_std_bidir<I2S: I2s>(
+        i2s: impl Peripheral<P = I2S> + 'd,
+        config: &config::StdConfig,
         bclk: impl Peripheral<P = impl InputPin + OutputPin> + 'd,
-        din: Option<impl Peripheral<P = impl InputPin> + 'd>,
+        din: impl Peripheral<P = impl InputPin> + 'd,
+        dout: impl Peripheral<P = impl OutputPin> + 'd,
         mclk: Option<impl Peripheral<P = impl InputPin + OutputPin> + 'd>,
         ws: impl Peripheral<P = impl InputPin + OutputPin> + 'd,
     ) -> Result<Self, EspError> {
-        let port = I2S::port();
-        let chan_cfg = config.channel_cfg.as_sdk(port);
-
-        let mut rx_chan_handle: i2s_chan_handle_t = null_mut();
-
-        // Safety: &chan_cfg is a valid pointer to an i2s_chan_config_t.
-        // rx and tx are out pointers.
-        unsafe { esp!(i2s_new_channel(&chan_cfg, null_mut(), &mut rx_chan_handle,))? };
-
-        if rx_chan_handle.is_null() {
-            panic!("Expected non-null rx channel handle");
-        }
-
-        // Allocate the internal channel struct.
-        let rx = I2sChannel::new(port as u8, rx_chan_handle);
-
-        // Create the channel configuration.
-        let dout: Option<PeripheralRef<'d, AnyIOPin>> = None;
-        let std_config: i2s_std_config_t = config.as_sdk(
-            bclk.into_ref(),
-            din.map(|d_in| d_in.into_ref()),
-            dout,
-            mclk.map(|m_clk| m_clk.into_ref()),
-            ws.into_ref(),
-        );
-
-        // Safety: rx.chan_handle is a valid, non-null i2s_chan_handle_t,
-        // and &std_config is a valid pointer to an i2s_std_config_t.
-        unsafe {
-            // Open the RX channel.
-            esp!(i2s_channel_init_std_mode(rx.chan_handle, &std_config))?;
-        }
-
-        Ok(Self {
-            i2s: port as u8,
-            rx: Some(rx),
-            tx: None,
-            _p: PhantomData,
-            _dir: PhantomData,
-        })
+        Self::internal_new_std(
+            i2s,
+            config,
+            true,
+            true,
+            bclk,
+            Some(din),
+            Some(dout),
+            mclk,
+            ws,
+        )
     }
+}
 
-    #[cfg(esp_idf_version_major = "4")]
+impl<'d> I2sDriver<'d, I2sRx> {
+    /// Create a new standard mode driver for the given I2S peripheral with only the receive channel open.
     #[allow(clippy::too_many_arguments)]
-    pub fn new_rx<I2S: I2s>(
-        _i2s: impl Peripheral<P = I2S> + 'd,
-        config: config::StdConfig,
+    pub fn new_std_rx<I2S: I2s>(
+        i2s: impl Peripheral<P = I2S> + 'd,
+        config: &config::StdConfig,
         bclk: impl Peripheral<P = impl InputPin + OutputPin> + 'd,
-        din: Option<impl Peripheral<P = impl InputPin> + 'd>,
+        din: impl Peripheral<P = impl InputPin> + 'd,
```

Comment:
Similar to my comment on the TDM, there was a reason I had this an an Option<>, likely related to unfinished ADC work that is unlikely to come to fruition.